lib/puppet/provider/user/directoryservice.rb

```diff
@@ -619,13 +619,4 @@ def set_salted_pbkdf2(users_plist, shadow_hash_data, field, value)
   def write_users_plist_to_disk(users_plist)
     Puppet::Util::Plist.write_plist_file(users_plist, "#{users_plist_dir}/#{@resource.name}.plist", :binary)
   end
-
-  # This is a simple wrapper method for writing values to a file.
-  def write_to_file(filename, value)
```

Comment:
You could make a case for that, and usually I would.  However, in this case, I think we're OK.

* `write_to_file` was introduced by Gary / Jeff back in 2012 https://github.com/puppetlabs/puppet/commit/960ab6b03e32ed770ecdab4bf2bc269d532b8001 and was called by `write_sha1_hash` which in turn was called by the `password=` setter
* In https://github.com/Iristyle/puppet/commit/efd2a45646eaedd3aa744ebcf7beb94739653db5 / [PUP-3553](https://tickets.puppetlabs.com/browse/PUP-3553) the `password=` setter was completely rewritten when we dropped support for OSX < 10.6.  It appears that this `write_to_file` method was accidentally left in at that point, as it no longer had any callers - see full file at https://github.com/Iristyle/puppet/blob/efd2a45646eaedd3aa744ebcf7beb94739653db5/lib/puppet/provider/user/directoryservice.rb
* You could argue that this is still public API - however, this a very specific provider implementation on OSX, so while not impossible, I would be very surprised if anyone was using it.  Specs used to use it prior to PUP-3553, but its usage there was removed as well.

If you feel really strongly that we should deprecate this method to be safe, we can do that too.

I should probably update commit message with the above info regardless of whether we decide on removing or deprecating / removing in 5.